python/kubeflow/trainer/utils/utils.py

```diff
@@ -121,20 +190,19 @@ def get_runtime_trainer(
     if not (trainer_container and trainer_container.image):
         raise Exception(f"Runtime doesn't have trainer container {replicated_jobs}")
 
-    # Extract image name from the container image to get appropriate Trainer.
-    image_name = trainer_container.image.split(":")[0]
-    trainer = types.ALL_TRAINERS.get(image_name, types.DEFAULT_TRAINER)
+    # Use the new detection logic with fallback
+    trainer = _detect_trainer(trainer_container)
```

Comment:
Do we really need to use `_` prefix for the util functions ?
E.g. we have many other internal util functions where we don't have such prefix.